src/libAtomVM/nifs.c

```diff
@@ -1284,12 +1284,40 @@ static term nif_erlang_spawn(Context *ctx, int argc, term argv[])
 static term nif_erlang_send_2(Context *ctx, int argc, term argv[])
 {
     UNUSED(argc);
+    term dest = argv[0];
+    GlobalContext *glb = ctx->global;
+    struct ListHead *processes_table_list = synclist_rdlock(&glb->processes_table);
+
+    Context *p = NULL;
+    struct ListHead *item;
+    if (term_is_pid(dest)) {
+        int32_t dest_pid = term_to_local_process_id(dest);
+
+        globalcontext_send_message_nolock(glb, dest_pid, argv[1]);
+        synclist_unlock(&glb->processes_table);
 
-    term pid_term = argv[0];
-    VALIDATE_VALUE(pid_term, term_is_pid);
+    } else if (term_is_atom(dest)) {
+        int atom_index = term_to_atom_index(dest);
+
+        int test_pid = globalcontext_get_registered_process(glb, atom_index);
+        if (UNLIKELY(test_pid == 0)) {
+            synclist_unlock(&glb->processes_table);
+            RAISE_ERROR(BADARG_ATOM);
+        }
 
-    int local_process_id = term_to_local_process_id(pid_term);
-    globalcontext_send_message(ctx->global, local_process_id, argv[1]);
+        p = globalcontext_get_process_nolock(glb, test_pid);
```

Comment:
also let's use a more meaningful name, such as `target`, or `registered_process` or anything like that.